src/iiiftilesource.js

```diff
@@ -492,10 +517,10 @@ $.extend( $.IIIFTileSource.prototype, $.TileSource.prototype, /** @lends OpenSea
                 iiifSize = "full";
             } else if ( this.version === 3 && iiifSizeW === this.width && iiifSizeH === this.height ) {
                 iiifSize = "max";
-            } else if (this.version === 3) {
-                iiifSize = iiifSizeW + "," + iiifSizeH;
-            } else {
+            } else if (this.isLevel0) {
```

Comment:
Does this need a check to see which version is being asked for? The width only is only valid for v2 and v2.1 (https://iiif.io/api/image/2.0/change-log/#added-canonical-uri-form)

v3 needs both width and size. Version 1 doesn't specify a canonical form.

Reply:
Well, mathematically, both the width and height are necessary to be able to get a tile at the right size, so strict canonicality has to be sacrificed if we want OSD to work with these kinds of images with version 2.x of the API. Width,height requests are a requirement for a level 1 server, so in theory, this change should not break any existing IIIF setup.

Reply:
I may have missread the code but I was thinking this change would break a level0 version 3 server which expects width,height. If the test was:
```
} else if (this.isLevel0 && this.version === 2) {
```
then the v3 level 0 use case would fall to the else which asks for both width and height

Reply:
OK, sorry I see what you mean. What about v1? You say it doesn't specify a canonical form, but what do static tilers like iiif-tiler actually produce for v1 level0? Just width like v2? If so, then maybe we need something like this

`} else if (this.isLevel0 && this.version < 3) {`

Reply:
OK, applied in commit https://github.com/openseadragon/openseadragon/pull/2710/commits/c13480c2e3190970a73244015530e729b105b856